File: app/src/main/java/com/mdgd/pokemon/ui/splash/SplashViewModelFactory.java
package com.mdgd.pokemon.ui.splash;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.mdgd.pokemon.models.AppModule;

public class SplashViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final AppModule appComponent;

    public SplashViewModelFactory(AppModule appComponent) {
        super();
        this.appComponent = appComponent;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass == SplashViewModel.class) {
            return (T) new SplashViewModel(appComponent.getCache());
        }
        return null;
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/splash/SplashViewModelFactory.kt
package com.mdgd.pokemon.ui.splash

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.mdgd.pokemon.models.AppModule

class SplashViewModelFactory(private val appComponent: AppModule) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return if (modelClass == SplashViewModel::class.java) {
            SplashViewModel(appComponent.getCache()) as T
        } else super.create(modelClass)
    }
}
